src/coreclr/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -1498,15 +1498,15 @@ internal static void SafeHandleRelease(SafeHandle pHandle)
         internal static extern IntPtr GetCOMIPFromRCW(object objSrc, IntPtr pCPCMD, out IntPtr ppTarget, out bool pfNeedsRelease);
 #endif // FEATURE_COMINTEROP
 
+#if PROFILING_SUPPORTED
         //-------------------------------------------------------
         // Profiler helpers
         //-------------------------------------------------------
-#if PROFILING_SUPPORTED
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern IntPtr ProfilerBeginTransitionCallback(IntPtr pSecretParam, IntPtr pThread, object pThis);
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "StubHelpers_ProfilerBeginTransitionCallback")]
```

Comment:
What is a QCall?

Is the idea that this attribute will lead to `PopulateNDirectMethodDescImpl` passing [the condition to set the IsQCall property](https://github.com/dotnet/runtime/blob/094649e26a18cd3cdedc7e93a0efabb3da4156b2/src/coreclr/vm/dllimport.cpp#L4396-L4399), so once `CreateNDirectStubAccessMetadata` is hit, the flag bit is toggled and we will always skip notifying the profiler of transitions? Or is it something else that dictates that condition?

It looks like `SF_SkipTransitionNotify` will overrule `CORProfilerTrackTransitions`, do we need `CORProfilerTrackTransitions`? I'm missing the context of when `PopulateNDirectMethodDescImpl` and `CreateNDirectStubAccessMetadata` are invoked, but what are the scenarios in which the `NDIRECTSTUB_FL_SKIP_TRANSITION_NOTIFY` bit won't be set?

Reply:
A QCall is a normal P/Invoke into the runtime (for example. `coreclr.dll` binary). If you look in `ProfilerManagedToUnmanagedTransitionMD`, you will notice that we don't notify the profiler for QCalls anyways. This change is designed to not even emit the code that is capable of doing the emit if the stub is for a QCall. It was previously unnecessary work that was being done.